frontend/public/components/service-instance.tsx

```diff
@@ -79,40 +79,58 @@ ServiceInstanceDetailsPage.displayName = 'ServiceInstanceDetailsPage';
 
 const ServiceInstancesHeader = props => <ListHeader>
   <ColHead {...props} className="col-md-3 col-sm-4 col-xs-6" sortField="metadata.name">Name</ColHead>
-  <ColHead {...props} className="col-md-3 col-sm-4 col-xs-6" sortField="metadata.namespace">Namespace</ColHead>
-  <ColHead {...props} className="col-md-2 col-sm-4 hidden-xs" sortFunc="serviceCatalogStatus">Status</ColHead>
-  <ColHead {...props} className="col-md-2 hidden-sm hidden-xs" sortFunc="planExternalName">Plan</ColHead>
+  <ColHead {...props} className="col-md-2 col-sm-3 col-xs-6" sortField="metadata.namespace">Namespace</ColHead>
+  <ColHead {...props} className="col-md-2 col-sm-3 hidden-xs" sortField="metadata.className">Service Class</ColHead>
+  <ColHead {...props} className="col-md-2 col-sm-2 hidden-xs" sortFunc="serviceCatalogStatus">Status</ColHead>
+  <ColHead {...props} className="col-md-1 hidden-sm hidden-xs" sortFunc="planExternalName">Plan</ColHead>
   <ColHead {...props} className="col-md-2 hidden-sm hidden-xs" sortField="metadata.creationTimestamp">Created</ColHead>
 </ListHeader>;
 
-const ServiceInstancesRow: React.SFC<ServiceInstancesRowProps> = ({obj}) => <div className="row co-resource-list__item">
-  <div className="col-md-3 col-sm-4 col-xs-6 co-resource-link-wrapper">
-    <ResourceCog actions={menuActions} kind={ServiceInstancesReference} resource={obj} />
-    <ResourceLink kind={ServiceInstancesReference} name={obj.metadata.name} namespace={obj.metadata.namespace} title={obj.metadata.name} />
-  </div>
-  <div className="col-md-3 col-sm-4 col-xs-6 co-break-word">
-    <ResourceLink kind="Namespace" name={obj.metadata.namespace} title={obj.metadata.namespace} />
-  </div>
-  <div className="col-md-2 col-sm-4 hidden-xs co-break-word">
-    <StatusWithIcon obj={obj} />
-  </div>
-  <div className="col-md-2 hidden-sm hidden-xs co-break-word">
-    {planExternalName(obj) || '-'}
-  </div>
-  <div className="col-md-2 hidden-sm hidden-xs co-break-word">
-    <Timestamp timestamp={obj.metadata.creationTimestamp} />
-  </div>
-</div>;
+const ServiceInstancesRow: React.SFC<ServiceInstancesRowProps> = ({obj}) => {
+  return <div className="row co-resource-list__item">
+    <div className="col-md-3 col-sm-4 col-xs-6 co-resource-link-wrapper">
+      <ResourceCog actions={menuActions} kind={ServiceInstancesReference} resource={obj} />
+      <ResourceLink kind={ServiceInstancesReference} name={obj.metadata.name} namespace={obj.metadata.namespace} title={obj.metadata.name} />
+    </div>
+    <div className="col-md-2 col-sm-3 col-xs-6 co-break-word">
+      <ResourceLink kind="Namespace" name={obj.metadata.namespace} title={obj.metadata.namespace} />
+    </div>
+    <div className="col-md-2 col-sm-3 hidden-xs co-break-word">
+      <ResourceLink kind="ClusterServiceClass" displayName={obj.spec.clusterServiceClassExternalName} title={obj.spec.clusterServiceClassExternalName} name={obj.spec.clusterServiceClassRef.name} />
```

Comment:
Same comment here: `obj.spec.clusterServiceClassRef.name` is not always defined, so this could be a runtime error